yapf/yapflib/unwrapped_line.py

```diff
@@ -383,6 +383,11 @@ def _IsSurroundedByBrackets(tok):
 
 def _SplitPenalty(prev_token, cur_token):
   """Return the penalty for breaking the line before the current token."""
+
+  if prev_token.value == 'not' and (cur_token.value == 'True'
```

Comment:
I can confirm that this passes your tests as simply:

    if prev_token.value == 'not':

change this, squash and rebase. I think this is a nice enhancement!